### ISSUE TITLE
neo4j-admin full import requires a restart

### ISSUE THREAD

**Issue by jdreo** [neo4j/neo4j#13307]
When trying to `neo4j-admin database import full --overwrite-destination=true`, the call ends on an "import done" message, but no data is created yet. Only after a restart of the server can we query the data.

## Steps to reproduce

Run the following shell script:
```sh
#!/bin/sh

echo ":ID,id,preferred_id,:LABEL" > header.csv

echo "PTPN14,'PTPN14','hugo','BiologicalEntity|Entity'" > data.csv

neo4j-admin database import full --delimiter="," --array-delimiter="|" --quote="'" --overwrite-destination=true --nodes="header.csv,data.csv" test

cypher-shell --username neo4j --database test --password $(cat neo4j.pass) "MATCH (n) RETURN n;"
```

Produce the following output
```
Neo4j version: 5.11.0-SNAPSHOT
Importing the contents of these files into /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/data/databases/test:
Nodes:
  /home/nojhan/code/oncodashkb/bug/header.csv
  /home/nojhan/code/oncodashkb/bug/data.csv


Available resources:
  Total machine memory: 7.653GiB
  Free machine memory: 353.3MiB
  Max heap memory : 1.702GiB
  Max worker threads: 4
  Configured max memory: 180.7MiB
  High parallel IO: true

WARNING: heap size 1.702GiB is unnecessarily large for completing this import.
The abundant heap memory will leave less memory for off-heap importer caches. Suggested heap size is 1.000GiB
Import starting 2023-09-22 14:56:35.277+0200
  Estimated number of nodes: 1.00 
  Estimated number of node properties: 2.00 
  Estimated number of relationships: 0.00 
  Estimated number of relationship properties: 0.00 
  Estimated disk space usage: 61B
  Estimated required memory usage: 289.1MiB

(1/4) Node import 2023-09-22 14:56:35.288+0200
  Estimated number of nodes: 1.00 
  Estimated disk space usage: 61B
  Estimated required memory usage: 289.1MiB
-......... .......... .......... .......... ..........   5% ∆83ms
.......... .......... .......... .......... ..........  10% ∆2ms
.......... .......... .......... .......... ..........  15% ∆1ms
.......... .......... .......... .......... ..........  20% ∆0ms
.......... .......... .......... .......... ..........  25% ∆1ms
.......... .......... .......... .......... ..........  30% ∆0ms
.......... .......... .......... .......... ..........  35% ∆1ms
.......... .......... .......... .......... ..........  40% ∆1ms
.......... .......... .......... .......... ..........  45% ∆0ms
.......... .......... .......... .......... ..........  50% ∆0ms
.......... .......... .......... .......... ..........  55% ∆1ms
.......... .......... .......... .......... ..........  60% ∆0ms
.......... .......... .......... .......... ..........  65% ∆1ms
.......... .......... .......... .......... ..........  70% ∆0ms
.......... .......... .......... .......... ..........  75% ∆0ms
.......... .......... .......... .......... ..........  80% ∆1ms
.......... .......... .......... .......... ..........  85% ∆0ms
.......... .......... .......... .......... ..........  90% ∆0ms
.......... .......... .......... .......... ..........  95% ∆1ms
.......... .......... .......... .......... .......... 100% ∆0ms

Node import COMPLETED in 259ms

(2/4) Relationship import 2023-09-22 14:56:35.548+0200
  Estimated number of relationships: 0.00 
  Estimated disk space usage: 0B
  Estimated required memory usage: 296.8MiB
Relationship import COMPLETED in 36ms

(3/4) Relationship linking 2023-09-22 14:56:35.585+0200
  Estimated required memory usage: 289.1MiB
Relationship linking COMPLETED in 42ms

(4/4) Post processing 2023-09-22 14:56:35.627+0200
  Estimated required memory usage: 289.1MiB
-......... .......... .......... .......... ..........   5% ∆59ms
.......... .......... .......... .......... ..........  10% ∆0ms
.......... .......... .......... .......... ..........  15% ∆0ms
.......... .......... .......... .......... ..........  20% ∆1ms
.......... .......... .......... .......... ..........  25% ∆0ms
.......... .......... .......... .......... ..........  30% ∆0ms
.......... .......... .......... .......... ..........  35% ∆1ms
.......... .......... .......... .......... ..........  40% ∆0ms
.......... .......... .......... .......... ..........  45% ∆0ms
.......... .......... .......... .......... ..........  50% ∆0ms
.......... .......... .......... .......... ..........  55% ∆0ms
.......... .......... .......... .......... ..........  60% ∆1ms
.......... .......... .......... .......... ..........  65% ∆0ms
.......... .......... .......... .......... ..........  70% ∆0ms
.......... .......... .......... .......... ..........  75% ∆0ms
.......... .......... .......... .......... ..........  80% ∆0ms
.......... .......... .......... .......... ..........  85% ∆1ms
.......... .......... .......... .......... ..........  90% ∆0ms
.......... .......... .......... .......... ..........  95% ∆0ms
.......... .......... .......... .......... .......... 100% ∆0ms

Post processing COMPLETED in 691ms


IMPORT DONE in 1s 159ms. 
Imported:
  1 nodes
  0 relationships
  2 properties
Peak memory usage: 296.8MiB
+---+
| n |
+---+
+---+

0 rows
ready to start consuming query after 2 ms, results consumed after another 0 ms
```

### Expected behaviour

At this stage, the behaviour I would have expected is an operational import or an error message.

If the database server is stopped and restarted, the data appear:
```
cypher-shell --username neo4j --database test --password $(cat neo4j.pass) "MATCH (n) RETURN n;"
+------------------------------------------------------------------------------------------------+
| n                                                                                              |
+------------------------------------------------------------------------------------------------+
| (:Entity:ThingWithTaxon:Gene:BiologicalEntity:NamedThing {preferred_id: "hugo", id: "PTPN14"}) |
+------------------------------------------------------------------------------------------------+

1 row
ready to start consuming query after 177 ms, results consumed after another 9 ms
```

Either the [ import documentation page](https://neo4j.com/docs/operations-manual/current/tools/neo4j-admin/neo4j-admin-import/) should mention that a restart is needed, either this is a bug.

## System config

- Neo4j version: 5.11.0-SNAPSHOT, built locally, with PATH exported toward its root.
- Operating system: Ubuntu 22.04.3 LTS
- API/Driver: neo4j-admin database import full

### Detailed logs

The server log reads:
```
neo4j-admin server console
Directories in use:
home:         /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT
config:       /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/conf
logs:         /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/logs
plugins:      /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/plugins
import:       /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/import
data:         /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/data
certificates: /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/certificates
licenses:     /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/licenses
run:          /home/nojhan/apps/neo4j/packaging/standalone/target/neo4j-community-5.11.0-SNAPSHOT/run
Starting Neo4j.
2023-09-22 13:18:55.932+0000 INFO  Starting...
2023-09-22 13:18:57.206+0000 INFO  This instance is ServerId{48f50c8f} (48f50c8f-a7e0-4e48-8c4b-e88d27dbc91f)
2023-09-22 13:18:58.354+0000 INFO  ======== Neo4j 5.11.0-SNAPSHOT ========
2023-09-22 13:19:00.599+0000 INFO  Bolt enabled on 0.0.0.0:7687.
2023-09-22 13:19:01.462+0000 INFO  Remote interface available at http://localhost:7474/
2023-09-22 13:19:01.493+0000 INFO  id: C3D68D9FA1DB2E6608C44FDEAA924F9F88078349BD6FC7EF6FFC3D9CEB9C932D
2023-09-22 13:19:01.493+0000 INFO  name: system
2023-09-22 13:19:01.493+0000 INFO  creationDate: 2023-09-11T14:46:02.281Z
2023-09-22 13:19:01.494+0000 INFO  Started.
```

The `neo4j.conf`:
```
initial.dbms.default_database=test
server.directories.import=import
server.bolt.enabled=true
server.bolt.listen_address=0.0.0.0:7687
server.bolt.advertised_address=:7687
server.http.enabled=true
server.http.listen_address=:7474
server.http.advertised_address=:7474
server.https.enabled=false
server.https.listen_address=:7473
server.https.advertised_address=:7473
server.jvm.additional=-XX:+UseG1GC
server.jvm.additional=-XX:-OmitStackTraceInFastThrow
server.jvm.additional=-XX:+AlwaysPreTouch
server.jvm.additional=-XX:+UnlockExperimentalVMOptions
server.jvm.additional=-XX:+TrustFinalNonStaticFields
server.jvm.additional=-XX:+DisableExplicitGC
server.jvm.additional=-Djdk.nio.maxCachedBufferSize=1024
server.jvm.additional=-Dio.netty.tryReflectionSetAccessible=true
server.jvm.additional=-Djdk.tls.ephemeralDHKeySize=2048
server.jvm.additional=-Djdk.tls.rejectClientInitiatedRenegotiation=true
server.jvm.additional=-XX:FlightRecorderOptions=stackdepth=256
server.jvm.additional=-XX:+UnlockDiagnosticVMOptions
server.jvm.additional=-XX:+DebugNonSafepoints
server.jvm.additional=--add-opens=java.base/java.nio=ALL-UNNAMED
server.jvm.additional=--add-opens=java.base/java.io=ALL-UNNAMED
server.jvm.additional=--add-opens=java.base/sun.nio.ch=ALL-UNNAMED
server.jvm.additional=-Dlog4j2.disable.jmx=true
server.windows_service_name=neo4j
```

**Comment by emmaholmbergohlsson** [comment]
Hello @jdreo , thank you for reporting this.
No you should not have to restart the whole database server. What you should have to do is start the imported database.

You did discover a bug though :tada:  - import didn't block you from importing to an online database like it should have. I'll fix that, so full import correctly blocks import to online databases.
If you change your script to first stop the database, then do the import, then start the database, and lastly do your query, it should work as expected

**Comment by jdreo** [comment]
I see, I misunderstood how the import worked. But an error message would have helped, indeed. Glad to see my mistake helped, if just a little bit :smile:

**Comment by emmaholmbergohlsson** [comment]
Hey @jdreo , the fix is ready and will be included in the next release (5.14).
Can I close this issue or is there anything more you need help with here?

**Comment by jdreo** [comment]
It’s all good, thanks.
